src/AcceptanceTests/Infrastructure/Cleaner.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace NServiceBus.AzureServiceBus.AcceptanceTests.Infrastructure
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Logging;
+    using Microsoft.ServiceBus;
+    using Microsoft.ServiceBus.Messaging;
+    using NServiceBus.AcceptanceTests;
+    using NUnit.Framework;
+
+    public class Cleaner : NServiceBusAcceptanceTest
+    {
+        [Test, Explicit("Intended to be executed explicitly to delete all queues and topics.")]
+        [Category("Cleanup")]
+        public async Task DeleteEntities()
+        {
+            var namespaceManager = NamespaceManager.CreateFromConnectionString(TestUtility.GetDefaultConnectionString());
+
+            var queryQueues = namespaceManager.GetQueuesAsync();
+            var queryTopics = namespaceManager.GetTopicsAsync();
+            await Task.WhenAll(queryQueues, queryTopics)
+                .ConfigureAwait(false);
+
+            var deleteQueues = queryQueues.Result.Select(queueDescription => namespaceManager.DeleteQueueAsync(queueDescription.Path));
+            var deleteTopics = queryTopics.Result.Select(topicDescription => namespaceManager.DeleteTopicAsync(topicDescription.Path));
+
+            await TryWithRetries(Task.WhenAll(deleteQueues.Concat(deleteTopics)), 5)
+                .ConfigureAwait(false);
+        }
+
+        static async Task TryWithRetries(Task task, int maxRetryAttempts, int usedRetryAttempts = 0)
+        {
+            try
+            {
+                await task
+                    .ConfigureAwait(false);
+            }
+            catch (Exception exception)
+                when (usedRetryAttempts < maxRetryAttempts && (exception is TimeoutException || exception is MessagingCommunicationException || exception is ServerBusyException))
+            {
+                logger.InfoFormat("Attempt to delete all queues and topics has failed. Trying attempt {0}/{1} in 5 seconds.", usedRetryAttempts + 2, maxRetryAttempts);
```

Comment:
Why is it here `usedRetryAttempts + 2`, below `usedRetryAttempts + 1` and further down `usedRetryAttempts` that makes it hard to read and understand

Reply:
When it's a first attempt, `usedRetryAttempts`  will be zero.
Writing "attempt 1/5" is meaningless. Next attempt is a second attempt, so it's "attempt 2/5 in 5 seconds". When we recursively invoke this method, only then `usedRetryAttempts` is increased and becomes 1. So you need to add 2 to get the message correct.